modules/core/include/opencv2/core/mat.hpp

```diff
@@ -165,7 +165,8 @@ class CV_EXPORTS _InputArray
         UMAT              =10 << KIND_SHIFT,
         STD_VECTOR_UMAT   =11 << KIND_SHIFT,
         STD_BOOL_VECTOR   =12 << KIND_SHIFT,
-        STD_VECTOR_CUDA_GPU_MAT = 13 << KIND_SHIFT
+        STD_VECTOR_CUDA_GPU_MAT = 13 << KIND_SHIFT,
+        STD_BOOL_VECTOR_VECTOR = 14 << KIND_SHIFT
```

Comment:
Addition of new kind should have implementation for almost all Input/Output array methods, not only getMat(int) - list is long here.
Also new core functionality should be covered by tests...